cmd/podman/save.go

```diff
@@ -165,12 +164,14 @@ func imageNameForSaveDestination(img *libpodImage.Image, imgUserInput string) st
 	}
 
 	prepend := ""
-	if !strings.Contains(imgUserInput, libpodImage.DefaultLocalRepo) {
+	if !strings.Contains(imgUserInput, libpodImage.DefaultLocalRegistry) {
 		// we need to check if localhost was added to the image name in NewFromLocal
 		for _, name := range img.Names() {
-			// if the user searched for the image whose tag was prepended with localhost, we'll need to prepend localhost to successfully search
-			if strings.Contains(name, libpodImage.DefaultLocalRepo) && strings.Contains(name, imgUserInput) {
-				prepend = fmt.Sprintf("%s/", libpodImage.DefaultLocalRepo)
+			// If the user is saving an image in the localhost registry,  getLocalImage need
+			// a name that matches the format localhost/<tag1>:<tag2> or localhost/<tag>:latest to correctly
+			// set up the manifest and save.
+			if strings.Contains(name, libpodImage.DefaultLocalRegistry) && strings.Contains(name, imgUserInput) {
```

Comment:
Wait, the rebase has somehow dropped the `Contains`→`HasPrefix` changes (in both places), and the requirement to match `localhost/` , not just `localhost`.

I mean, this PR as it it currently is definitely not making things _worse_, but fixing the “contains `localhost`” was AFAICT the very purpose of this PR.

Reply:
shoot good catch @mtrmac . I'll fix that.

Reply:
ACK, thanks for the quick fix!

(The comment is still not correct, but it was not correct before either, the code is better, and as I said I’ll defer on the comment to others,  who have approved this PR.)

Reply:
@mtrmac Yeah I only kind of understand the situation. If you have any suggestions on rewording I'd be happy to change it. This is my best interpretation of why that code is needed.

Reply:
As I understand the situation (which may not be correct, I don’t have tests for all of that) is 
https://github.com/projectatomic/libpod/pull/1140#discussion_r204933964 .  Really most of that behavior should be changed, so not documenting it in the meantime is fine (documenting it could make it seem intentional, when it’s almost certainly not) — hopefully my refactoring work will give me enough confidence to simplify and fix most of that.

Reply:
ok SGTM